easybuild/easyblocks/generic/systemcompiler.py

```diff
@@ -27,18 +27,48 @@
 
 @author Bernd Mohr (Juelich Supercomputing Centre)
 @author Kenneth Hoste (Ghent University)
+@author Alan O'Cais (Juelich Supercomputing Centre)
 """
 import os
 import re
+from vsc.utils import fancylogger
 
 from easybuild.easyblocks.generic.bundle import Bundle
+from easybuild.easyblocks.generic.intelbase import IntelBase
+from easybuild.easyblocks.g.gcc import EB_GCC
 from easybuild.tools.filetools import read_file, which
 from easybuild.tools.run import run_cmd
 from easybuild.framework.easyconfig.easyconfig import ActiveMNS
+from easybuild.framework.easyconfig import CUSTOM
 from easybuild.tools.build_log import EasyBuildError
 
+_log = fancylogger.getLogger('easyblocks.generic.systemcompiler')
 
-class SystemCompiler(Bundle):
+def extract_compiler_version(compiler_name):
+    """Extract compiler version for provided compiler_name."""
+    # look for 3-4 digit version number, surrounded by spaces
+    # examples:
+    # gcc (GCC) 4.4.7 20120313 (Red Hat 4.4.7-11)
+    # Intel(R) C Intel(R) 64 Compiler XE for applications running on Intel(R) 64, Version 15.0.1.133 Build 20141023
+    if compiler_name == 'gcc':
+        out, _ = run_cmd("gcc --version", simple=False)
+    elif compiler_name in ['icc', 'ifort']:
+        out, _ = run_cmd("%s -V" % compiler_name, simple=False)
```

Comment:
@geimer The version extracted here doesn't actually match the version used by EasyBuild. I could just extract that from the product path (for <2016 one search string, for later stuff another)